djangocms_versioning_filer/monkeypatch/admin/admin_fields.py

```diff
@@ -0,0 +1,187 @@
+
+
+def validate(self, value, model_instance):
+    from django.core import exceptions
+    from django.db import router
+    if self.remote_field.parent_link:
+        return
+    super(ForeignKey, self).validate(value, model_instance)
+    if value is None:
+        return
+
+    using = router.db_for_read(self.remote_field.model, instance=model_instance)
+    # we need to use the original manager to access drafts
+    qs = self.remote_field.model._original_manager.using(using).filter(
+        **{self.remote_field.field_name: value}
+    )
+    qs = qs.complex_filter(self.get_limit_choices_to())
+    if not qs.exists():
+        raise exceptions.ValidationError(
+            self.error_messages['invalid'],
+            code='invalid',
+            params={
+                'model': self.remote_field.model._meta.verbose_name, 'pk': value,
+                'field': self.remote_field.field_name, 'value': value,
+            },  # 'pk' is included for backwards compatibility
+        )
+
+
+filer.fields.file.AdminFileWidget.render = render
+filer.fields.file.AdminFileWidget.obj_for_value = obj_for_value
+filer.fields.file.AdminFileFormField.__init__ = __init__
+filer.fields.file.FilerFileField.validate = validate
+
+
+@receiver(post_version_operation)
+def post_version_publish_image_versions(sender, **kwargs):
+    '''Triggered when a page is published. This is when any image plugins also
+    should become published'''
+    from collections import defaultdict
+
+    operation = kwargs['operation']
+    if operation != OPERATION_PUBLISH and sender is PageContent:
+        return
+
+    # import ipdb; ipdb.set_trace()
+
+    pagecontent = kwargs['obj'].content
+    placeholder = pagecontent.get_placeholders()[0]  # hacky way to get the component placeholder.
+
+    # instead of doing placeholder get_plugins we could use the ids from placeholder.
+    plugin_instances = placeholder.get_plugins()
+    future_instances = defaultdict(list)
+    for plugin_instance in plugin_instances:
+        plugin_pk = plugin_instance.pk
+        Model = plugin_instance.get_plugin_class().model
+        fields = Model._meta.fields
+
+        for field in fields:
+            if isinstance(field, FilerFileField):
+                field_name = field.name
+                future_instances[(Model, field_name)].append(plugin_pk)
+
+    image_attrs = []
+    for (Model, field_name), pks in future_instances.items():
+        image_attrs = [
+            getattr(inst, field_name)
+            for inst in Model.objects.filter(cmsplugin_ptr__in=pks)
+        ]
+        image_attrs.extend(image_attrs)
+
+    for instance in image_attrs:
+        # fetching all versions by a grouper_id would be better. instance.field_name.grouper_id, most efficient.
+        latest_version = Version.objects.for_grouper(instance.grouper).first()
+        if latest_version.state == DRAFT:
+            latest_version.publish()
```

Comment:
I don't remember exactly how things were handled in headless. The question is, is this approach compatible with how things would work in headless mode?

Reply:
Removed.